### ISSUE TITLE
Support for Formatted Validation Messages in Textbox Component

### ISSUE THREAD

**Issue by hagad** [Sage/carbon#7388]
### Description

As a developer, I would like the ability to display formatted error messages, especially for longer messages.
Currently, the validation prop on the Carbon Textbox component only accepts a plain string.
However, we have a need to support rich or formatted validation messages—for example, to emphasize certain parts of the message or improve readability for longer content..

Example of formatted error message: 
```
Message error
- Detail 1
- Detail 2
```

### Suggested solution

The error prop from [Carbon Textbox](https://carbon.sage.com/?path=/docs/textbox--docs#validation-states) currently only accepts a string.
It would be beneficial if the component could also accept a JSX.Element, allowing us to use SageMarkdown for richer and more readable validation messages.

### Confidentiality

- [x] I confirm there is no confidential or commercially sensitive information included.

**Comment by edleeks87** [comment]
@ljemmo @harpalsingh

**Comment by nicktitchmarsh** [comment]
@ljemmo This refers to error messages above inputs, allowing a React Node would make it so developers could pass in any HTML or components they want. Are you sure this is advisable? Readability and Accessibility should be considered as there may be unintended impact for screen reader users, the responsibility would be on the developers to make sure this is not breaking WCAG guidelines.
